### PR TITLE
Give different names to pools in PooledExecutionServiceTest and make the test reliable

### PR DESCRIPTION
**Pull request by henri-tremblay**
It contains an improved javadoc and better test code that can be permanent but that will also be helpful to diagnose with the test is failing

### CODE REVIEW

**Comment by henri-tremblay** [comment]
test this please

**Comment by henri-tremblay** [comment]
test this please

File: impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java [hunk 1]
```diff
@@ -113,7 +113,9 @@ public void start(ServiceProvider<Service> serviceProvider) {
   }
 
   /**
-   * Stop the service. Underlying executors will be stopped calling {@code shutdownNow}. Pending tasks are discarded
+   * Stop the service. Underlying executors will be stopped calling {@code shutdownNow}. Pending tasks are discarded. Currently running tasks are
+   * awaited for termination for 30 seconds. So it is possible for get out of this method after 30 seconds having tasks that are still running. If it
```
Comment (anthonydahanne) [inline impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java]: possible to get out ?

File: impl/src/test/java/org/ehcache/impl/internal/executor/PooledExecutionServiceTest.java [hunk 2]
```diff
@@ -29,8 +29,10 @@
 import java.util.Set;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
```
Comment (anthonydahanne) [inline impl/src/test/java/org/ehcache/impl/internal/executor/PooledExecutionServiceTest.java]: unused import

File: impl/src/test/java/org/ehcache/impl/internal/executor/PooledExecutionServiceTest.java [hunk 3]
```diff
@@ -29,8 +29,10 @@
...
 import java.util.concurrent.LinkedBlockingDeque;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
```
Comment (anthonydahanne) [inline impl/src/test/java/org/ehcache/impl/internal/executor/PooledExecutionServiceTest.java]: unused import

File: impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java (hunk 1 above)
Comment (henri-tremblay) [inline impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java]: yep thanks

File: impl/src/test/java/org/ehcache/impl/internal/executor/PooledExecutionServiceTest.java (hunk 2 above)
Comment (henri-tremblay) [inline impl/src/test/java/org/ehcache/impl/internal/executor/PooledExecutionServiceTest.java]: indeed

File: impl/src/test/java/org/ehcache/impl/internal/executor/PooledExecutionServiceTest.java (hunk 3 above)
Comment (henri-tremblay) [inline impl/src/test/java/org/ehcache/impl/internal/executor/PooledExecutionServiceTest.java]: indeed

**Comment by henri-tremblay** [comment]
Last build failed and gave me the information I wanted. So I will fix it in this PR.

File: impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java [hunk 4]
```diff
@@ -129,14 +130,32 @@ public void stop() {
       }
     }
     try {
-      if(!scheduledExecutor.awaitTermination(30, SECONDS)) {
-        LOGGER.warn("Timeout while waiting on scheduler to finish");
+      while(!scheduledExecutor.awaitTermination(30, SECONDS)) {
```
Comment (chrisdennis) [inline impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java]: Was this accidental, or do we really want to wait forever?

Comment (henri-tremblay) [inline impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java]: Good question. First, I was waiting for 30 seconds and that's it. Then I started thinking that getting out of the wait with task still running means we get out of `stop()` in some unknown weird state that is probably hiding a bug. So I ended up preferring to freeze the stop and log. From there, a thread dump will easily allow to debug the issue.

Comment (chrisdennis) [inline impl/src/main/java/org/ehcache/impl/internal/executor/PooledExecutionService.java]: That makes sense... and looking at the other executor shutdown paths seems to be consistent with existing behavior.

**Pull request merged**